tests/test_datasets.py

```diff
@@ -747,5 +749,79 @@ def test_scanpaths_from_fixations(fixation_indices):
     compare_fixations(sub_fixations, new_sub_fixations, crop_length=True)
 
 
+@pytest.mark.parametrize('attribute_name , attribute_value, invert_match', [('dva',1,False), ('some_strings','n',True), ('some_strings','b',True)])
+def test_filter_stimuli_by_attribute(file_stimuli_with_attributes, fixation_trains, attribute_name, attribute_value, invert_match):
+    new_fixations = pysaliency.Fixations.concatenate((fixation_trains, fixation_trains, fixation_trains[:2]))
+    new_fixations = new_fixations[:]
+    _stimuli, _fixations = filter_stimuli_by_attribute(file_stimuli_with_attributes, new_fixations, attribute_name, attribute_value, invert_match)
+    if attribute_name == 'dva' and attribute_value == 1 and invert_match is False:
+        _fixations2 = new_fixations[[5,6,7,13,14,15]]
+        compare_fixations(_fixations,_fixations2)
+        assert list(_stimuli.stimulus_ids) == file_stimuli_with_attributes.stimulus_ids[[1]]
+    if attribute_name == 'some_strings' and attribute_value == 'n' and invert_match is True:
+        _fixations2 = new_fixations[list(range(0,18))]
+        compare_fixations(_fixations,_fixations2)
+        assert list(_stimuli.stimulus_ids) == file_stimuli_with_attributes.stimulus_ids[list(range(0,13))+list(range(14,18))]
+    if attribute_name == 'some_strings' and attribute_value == 'b' and invert_match is True:
+        _fixations2 = new_fixations[[0,1,2,3,4,8,9,10,11,12,16,17]]
+        compare_fixations(_fixations,_fixations2)
+        assert list(_stimuli.stimulus_ids) == file_stimuli_with_attributes.stimulus_ids[list(range(0,1))+list(range(2,18))]
+
+
+@pytest.mark.parametrize('attribute_name , attribute_value, invert_match', [('subjects',0, True),('some_attribute',2, False), ('some_attribute',3, False),  ('some_attribute',4, False)])
+def test_filter_fixations_by_attribute(fixation_trains, attribute_name, attribute_value, invert_match):
+    fixations = fixation_trains[:]
+    _fixations1 = filter_fixations_by_attribute(fixations, attribute_name, attribute_value, invert_match)
+    if attribute_name == 'subjects' and attribute_value == 0 and invert_match is True:
+        _fixations2 = fixations[[3,4,5,6,7]]
+        compare_fixations(_fixations1,_fixations2)
```

Comment:
blanks are missing quite often:
```suggestion
        compare_fixations(_fixations1, _fixations2)
```